### PR TITLE
SOLR-15182: replica API conversion to annotations

### PR DESCRIPTION
**Pull request by gerlowskija**
# Description

The v2 API currently has APIs implemented two ways: using an JSON 'apispec' file, or using a JAX-RS -like annotation framework.  The former framework is deprecated and we're in the process of moving APIs over to the latter, as it generally results in less duplication and has some other advantages.

# Solution

This commit rewrites the lone replica API ("DeleteReplicaAPI") to use the annotation-based approach.

# Tests

See V2ShardsAPIMappingTest

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [x] I have added tests for my changes.

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/handler/admin/CollectionsHandler.java [hunk 1]
```diff
@@ -20,48 +20,26 @@
...
 import org.apache.solr.client.solrj.impl.HttpSolrClient;
 import org.apache.solr.client.solrj.impl.HttpSolrClient.Builder;
 import org.apache.solr.client.solrj.request.CollectionAdminRequest;
 import org.apache.solr.client.solrj.request.CoreAdminRequest.RequestSyncShard;
 import org.apache.solr.client.solrj.response.RequestStatusState;
 import org.apache.solr.client.solrj.util.SolrIdentifierValidator;
-import org.apache.solr.cloud.OverseerSolrResponse;
-import org.apache.solr.cloud.OverseerSolrResponseSerializer;
-import org.apache.solr.cloud.OverseerTaskQueue;
+import org.apache.solr.cloud.*;
```
Comment (epugh) [inline solr/core/src/java/org/apache/solr/handler/admin/CollectionsHandler.java:30]: Nit pick that we don't do these ;-)   Where is our java linter ;-)

File: solr/core/src/java/org/apache/solr/handler/admin/CollectionsHandler.java [hunk 2]
```diff
@@ -20,48 +20,26 @@
...
 import org.apache.solr.common.cloud.Replica.State;
-import org.apache.solr.common.cloud.Slice;
-import org.apache.solr.common.cloud.SolrZkClient;
-import org.apache.solr.common.cloud.ZkCmdExecutor;
-import org.apache.solr.common.cloud.ZkCoreNodeProps;
-import org.apache.solr.common.cloud.ZkNodeProps;
-import org.apache.solr.common.cloud.ZkStateReader;
-import org.apache.solr.common.params.CollectionAdminParams;
-import org.apache.solr.common.params.CollectionParams;
+import org.apache.solr.common.params.*;
```
Comment (epugh) [inline solr/core/src/java/org/apache/solr/handler/admin/CollectionsHandler.java:41]: lint!

**Review by epugh — APPROVED** [review]
Looks good, nice to see some clean ups.   Lots of imports that end in `.*`

**Comment by gerlowskija** [comment]
Hmm, yeah, looks like I've got some IntelliJ editor settings to figure out.  Is there a settings file we have to import or something, or is it just manual settings people have formed consensus on?

**Comment by epugh** [comment]
I checked the `/dev-docs/` folder, assumming there might be something...   Readme says:

```
- *IntelliJ* - IntelliJ idea can import the project out of the box. 
               Code formatting conventions should be manually adjusted. 
```

**Pull request merged**